main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala

```diff
@@ -33,16 +34,16 @@ object MagicMatchCompleter {
     *   case Blue => ???
     * }
     */
-  def getCompletions(name: String, tpe: Type ,ctx: CompletionContext)(implicit root: TypedAst.Root): Iterable[Completion] = {
-    if (!"match".startsWith(name)) return Nil
-
-    val ident = extractIdentifier(ctx.word)
+  def getCompletions(tpe: Type ,ctx: CompletionContext)(implicit root: TypedAst.Root): Iterable[Completion] = {
+    val ident = extractIdentifier(ctx)
+    val range = extractRange(ctx)
```

Comment:
Instead of thinking in strings, we should try to rely on the source location in the FieldNotFound error. 

Then we can simply insert "match" right before that location and insert "{ case .." right after.

Reply:
If it helps, the goal is very much to __avoid__ lots of string computations.